test/test_cost.py

```diff
@@ -7144,6 +7184,9 @@ def test_reinforce_value_net(
             elif td_est is not None:
                 loss_fn.make_value_estimator(td_est)
             loss_td = loss_fn(td)
+            if reduction == "none":
+                assert loss_td.batch_size == td.batch_size
+                loss_td = loss_td.apply(lambda x: x.float().mean(), batch_size=[])
```

Comment:
maybe here instead of converting to float we can do
```python
def func(x):
    if x.dtype != torch.float:
        return
    return x.mean()
loss_td = loss_td.apply(func, batch_size=[])
```
because I suspect we convert to float things that we may not want to convert to float (eg metadata)